commit 3543364ddfa2c586029405cdaa8511c394739dfd
Author: Yongcheng Yang <yongcheng.yang@gmail.com>
Date:   Fri Dec 11 11:01:17 2015 -0500

    exportfs: Deal with path's trailing "/" in unexportfs_parsed()
    
    When unexport directory, it's possible that the specified path ends with
    a '/'. So we need to deal with it to find the matched entry.
    
    If not, there will be error like "Could not find '*:/some_path/' to
    unexport."
    
    Signed-off-by: Yongcheng Yang <yongcheng.yang@gmail.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/exportfs/exportfs.c b/utils/exportfs/exportfs.c
index acbd9e0..48a096c 100644
--- a/utils/exportfs/exportfs.c
+++ b/utils/exportfs/exportfs.c
@@ -376,8 +376,17 @@ unexportfs_parsed(char *hname, char *path, int verbose)
 			hname = ai->ai_canonname;
 	}
 
+	/*
+	 * It's possible the specified path ends with a '/'. But
+	 * the entry from exportlist won't has the trailing '/',
+	 * so need to deal with it.
+	*/
+	size_t nlen = strlen(path);
+	while (path[nlen - 1] == '/')
+		nlen--;
+
 	for (exp = exportlist[htype].p_head; exp; exp = exp->m_next) {
-		if (path && strcmp(path, exp->m_export.e_path))
+		if (path && strncmp(path, exp->m_export.e_path, nlen))
 			continue;
 		if (htype != exp->m_client->m_type)
 			continue;
